File: ui/src/main/java/com/walhalla/abcsharedlib/SharedNetwork.java
package com.walhalla.abcsharedlib;

public class SharedNetwork {

    public static final String PACKAGE_LIKEE = "video.like";
    public static final String PACKAGE_TWITTER = "com.twitter.android";
    public static final String PACKAGE_TWITTER_LITE = "com.twitter.android.lite";

    public static final String PACKAGE_INSTAGRAM = "com.instagram.android";
    public static final String PACKAGE_WHATSAPP = "com.whatsapp";
    public static final String PACKAGE_VIBER = "com.viber.voip";

    public static final String PACKAGE_FACEBOOK = "com.facebook.katana";
    public static final String PACKAGE_TIKTOK_M_PACKAGE = "com.zhiliaoapp.musically";
    public static final String PACKAGE_TIKTOK_T_PACKAGE = "com.ss.android.ugc.trill";

    public static final String PACKAGE_PINTEREST = "com.pinterest";


    public static String[] twitter() {
        return new String[]{
                SharedNetwork.PACKAGE_TWITTER,
                SharedNetwork.PACKAGE_TWITTER_LITE
        };
    }

    public static String[] pinterest() {
        return new String[]{
                SharedNetwork.PACKAGE_PINTEREST
        };
    }
}

File: shared/src/main/java/com/walhalla/abcsharedlib/SharedNetwork.kt
package com.walhalla.abcsharedlib

object SharedNetwork {
    fun twitter(): Array<String> {
        return arrayOf(
            Package.TWITTER, Package.TWITTER_LITE
        )
    }

    fun pinterest(): Array<String> {
        return arrayOf(
            Package.PINTEREST, Package.PINTERESTLITE
        )
    }

    object Package {
        const val LIKEE: String = "video.like"
        const val TWITTER: String = "com.twitter.android"
        const val TWITTER_LITE: String = "com.twitter.android.lite"

        const val INSTAGRAM: String = "com.instagram.android"
        const val WHATSAPP: String = "com.whatsapp"
        const val VIBER: String = "com.viber.voip"

        const val FACEBOOK: String = "com.facebook.katana"
        const val FACEBOOKLITE: String = "com.facebook.lite"


        const val TIKTOK_M_PACKAGE: String = "com.zhiliaoapp.musically"
        const val TIKTOK_LITE: String = "com.zhiliaoapp.musically.go"

        const val TIKTOK_T_PACKAGE: String = "com.ss.android.ugc.trill"


        const val PINTEREST: String = "com.pinterest"
        const val PINTERESTLITE: String = "com.pinterest.twa"
        const val RUOKANDROID: String = "ru.ok.android"
        const val REDDIT: String = "com.reddit.frontpage"

        const val YOUTUBE: String = "com.google.android.youtube"


        // com.google.android.youtube
        var tube: IntArray = intArrayOf(
            101,
            98,
            117,
            116,
            117,
            111,
            121,
            46,
            100,
            105,
            111,
            114,
            100,
            110,
            97,
            46,
            101,
            108,
            103,
            111,
            111,
            103,
            46,
            109,
            111,
            99
        )


        const val TRILLER: String = "co.triller.droid"
        const val KWAI: String = "com.kwai.video"
        const val RUOKLIVE: String = "ru.ok.live"
    }
}